### PR TITLE
sig-storage: update unit tests to use runStrategy instead of spec.Running

### PR DESCRIPTION
**Pull request by ShellyKa13**
### What this PR does
We want to deprecate the use of spec.Running and use spec.RunStrategy instead.
Before this PR: using spec.Running in storage UT

After this PR: using spec.RunStrategy in storage UT

Fixes # Jira-ticket: https://issues.redhat.com/browse/CNV-45439
relates to issue: #11993 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/storage/snapshot/snapshot_test.go [hunk 1]
```diff
@@ -748,7 +747,7 @@ var _ = Describe("Snapshot controlleer", func() {
 			It("should (partial) lock source if running", func() {
 				vmSnapshot := createVMSnapshotInProgress()
 				vm := createVM()
-				vm.Spec.Running = &t
+				vm.Spec.RunStrategy = pointer.P(v1.RunStrategyAlways)
```
Comment (alicefr) [inline pkg/storage/snapshot/snapshot_test.go:750]: aren't `f` and `t` used anymore? I don't see here their removal

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 2]
```diff
@@ -185,9 +185,6 @@ var _ = Describe("Clone", func() {
 		)
 		sourceVMI.Namespace = metav1.NamespaceDefault
 		sourceVM = libvmi.NewVirtualMachine(sourceVMI)
-		sourceVM.Spec.Running = nil
-		runStrategy := virtv1.RunStrategyHalted
-		sourceVM.Spec.RunStrategy = &runStrategy
```
Comment (alicefr) [inline pkg/virt-controller/watch/clone/clone_test.go:190]: Don't we need to set the RunStrategy to `Halted`. Is `nil` equivalent?

File: pkg/storage/snapshot/snapshot_test.go (hunk 1 above)
Comment (ShellyKa13) [reply pkg/storage/snapshot/snapshot_test.go:750]: this is used for ReadyToUse = &t. But I dont mind replacing that with the pointer like I did in other file

File: pkg/virt-controller/watch/clone/clone_test.go (hunk 2 above)
Comment (ShellyKa13) [reply pkg/virt-controller/watch/clone/clone_test.go:190]: It is already done in `sourceVM = libvmi.NewVirtualMachine(sourceVMI)`

File: pkg/storage/snapshot/snapshot_test.go (hunk 1 above)
Comment (ShellyKa13) [reply pkg/storage/snapshot/snapshot_test.go:748]: @alicefr added another commit to clean all pointers to t and f

**Comment by alromeros** [comment]
Not storage related but since there are just a handful of test files that use this maybe you could reconsider changing them all: `virt-controller/watch/vm_test.go`, `vms-admitter_test.go` and some virtctl-related tests. wdyt?

**Comment by ShellyKa13** [comment]
@alromeros I wouldn't have mind doing so, but there is a task for that for someone else..

**Comment by alicefr** [comment]
@ShellyKa13 looks great!
/approve

**Pull request merged**